worlds/hades/Client.py

```diff
@@ -0,0 +1,409 @@
+        hades_settings_string += str(self.hades_slot_data["storesanity"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["initial_weapon"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["ignore_greece_deaths"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["fatesanity"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["hidden_aspectsanity"]) + "-"
+        hades_settings_string += str(self.polycosmos_version) + "-"
+        hades_settings_string += str(self.hades_slot_data["automatic_rooms_finish_on_hades_defeat"]) + "-"
+
+        #Codify in the string all the finishing conditions
+        hades_settings_string += str(self.hades_slot_data["hades_defeats_needed"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["weapons_clears_needed"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["keepsakes_needed"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["fates_needed"]) + "-"
+            
+        #Store the compact setting string
+        self.compact_setting_string = hades_settings_string
+
+    def request_location_to_item_dictionary(self):
+        self.creating_location_to_item_mapping = True
+        request = self.missing_locations_cache + self.checked_locations_cache
+        asyncio.create_task(self.send_msgs([{"cmd": "LocationScouts", "locations": request, "create_as_hint": 0}]))
+
+    async def create_location_to_item_dictionary(self, itemsdict):
+        locationItemMapping = ""
+        for networkitem in itemsdict:
+            locationItemMapping += self.clear_invalid_char(self.location_names.lookup_in_slot(networkitem.location)) \
+                + "--" + self.clear_invalid_char(self.player_names[networkitem.player]) + "--" \
+                + self.clear_invalid_char(self.item_names.lookup_in_slot(networkitem.item, networkitem.player)) + "||"
+            
+        subsume.Send(styx_scribe_send_prefix + "Location to Item Map:" + locationItemMapping)
+        self.creating_location_to_item_dictionary = False
+        subsume.Send(styx_scribe_send_prefix + "Data finished" + self.compact_setting_string)
+    
+    def clear_invalid_char(self, inputstring: str):
+        newstr = inputstring.replace("{", "")
+        newstr = newstr.replace("}", "")
+        return newstr
+
+    # ----------------- Package Management section ends --------------------------------
+
+    # ----------------- Hints from game section starts --------------------------------
+
+    async def send_location_hint_to_server(self, message):
+        if self.hades_slot_data["store_give_hints"] == 0:
+            return
+        split_array = message.split("-")
+        request = []
+        for location in split_array:
+            if len(location) > 0:
+                request.append(self.location_name_to_id[location])
+        asyncio.create_task(self.send_msgs([{"cmd": "LocationScouts", "locations": request, "create_as_hint": 2}]))
```

Comment:
I'll stop marking these now, but if you search `create_task`, you'll probably want to replace all of them.